### DISCUSSION TITLE
Meta_Hours_6 - Club-Mate-Refactor

### DISCUSSION THREAD

**Discussion by yangwao** [kodadot/nft-gallery#2576] [Meta-Hours]
Have a Club Mate

<img width=400px src=https://user-images.githubusercontent.com/5887929/158803161-18e28703-51c2-43a1-a18a-c411e0199ff4.png>

- new stuff, numbers etc 

look back on superior productivity December 2021, Januar/February 2022

![image](https://user-images.githubusercontent.com/5887929/158802144-d0a1fa19-832b-4221-a1c5-3b693b17a227.png)

![image](https://user-images.githubusercontent.com/5887929/158802130-c25cfde9-3945-407f-9230-40586e6c841d.png)

![image](https://user-images.githubusercontent.com/5887929/158802099-1ca9dd4c-88f9-487b-9644-0081fb010c31.png)

![image](https://user-images.githubusercontent.com/5887929/158802235-e6df11e8-21e2-47eb-aef5-8e764a109524.png)

![image](https://user-images.githubusercontent.com/5887929/158802243-9aedf815-67cb-47f3-9e15-47ca3a9d4673.png)

- It's really nice to see we are quite progressing light speed and with all coordination. Then we had to take some intensive founders to retreat to recalibrate what we want to do next and bit unplug. 

- We had the first snapshot of the leaderboard, where Darshana is leading chart https://github.com/kodadot/nft-gallery/blob/main/LEADERBOARD.md

### Writings
- see @XyloDrone and @cryptodamsky done amazing writings along with others and would be nice to see everyone can add their own piece of puzzle 🧩✍️ write bits or two about their insights, and thoughts on the progress of kodadot, what've done, what we can do better and how others can engage to be more inviting and welcoming culture to drive artist and collectors. Best to aim to drive more discussions around KodaDot to craft a vibrant community, not because of the possibility of future token and value creation, but because people can create and drive network effects where they can be in ownership of craft something bigger

- tl;dr we should incentivize long-form writing more, it could be part of our ambassador program, plus organizing events and unofficial news outlets and creating playbooks for driving more artists to our marketplace
- we already have few issues labeled with writing if you are looking for inspiration where would you put your thoughts https://github.com/kodadot/nft-gallery/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Awriting-%E2%9C%8D%EF%B8%8F

- I've recently [wrote how we are rewarding contributors to KodaDot](https://dev.to/yangwao/how-we-reward-contributors-to-open-source-nft-marketplace-29bn)

### Onboarding
As KodaDot gets to the final phase becoming quite mature would be around clock to spread the word about it in Dotsama (polkadot+kusama) circles and outside Dotsama, probably attracting people from Solana etc. I believe we have a lot of stuff already to offer for collectors and artists  (lots of analytics, charts & graphs, comfortable UX, plus upcoming KODA protocol) we are quite confident to step up our game to participate in more agile markets.

I guess we are quite unique approach being community-owned and engaging in positive-sum games, we can help others drive revenue to their collections by collabs and spotlighting them etc. Be creative here.

- how to bring new people -- an open topic for discussion
- campaigns -- want to do campaign, how others can help you and elaborate? Yes, we can fund these efforts.
- incentivize creators for their collections? what would be rules?

### Developer bounties and payouts
- gradually I will be giving away rights to people payout themselves with payout bot, between each other once fixed https://github.com/kodadot/nft-gallery/issues/2116 
- #2448 is sparked discussion about the locking issues, time out the issue and avoiding double work.  This would help us to better coordinate on bigger scale issues as KodaDot won't be just about UI but other side products we craft in stealth, it would help us better scale with more people looking to earn an extra buck and side hustle, with future options becoming full-time contributor with continuous payroll

### Sail route
From start, we had tons of issues that come from brain dumps and various opportunities and we've been pretty loose on how to structure development. Nowadays, with crossing 100-200+ pull requests on monthly basis, we need to add a bit of guidance and steering, what we want to see being merged into the upstream codebase. So on top of what we are building and which issue you choose to solve, **we will be highlighting** which will have a bit higher priority to be merged into the codebase. We are starting with names of cities for easier recognition and we choose to start with Scandinavian picks, hope you like it. Also, you can then suggest other cities :)

- https://github.com/kodadot/nft-gallery/issues/2622
- https://github.com/kodadot/nft-gallery/issues/2646

### Development & me

- As mentioned above, I'll step down from development to focus on other venues of MetaPrime family. I feel quite confident that @roiLeo @vikiival @prachi00 @kkukelka @petersopko are doing a great job despite having in trust all of you over the last few weeks despite new talent is flowing to KodaDot I'll be observing from a close distance and sometimes helping with testing, giving alternative opinions etc. I guess it's a major stepping stone to how we operate and function. Plus it would open me more room to focus again on Ops and other non-tech stuff we have on plate and need process. I'll be stepping back and driving conceptually what KodaDot can be in the future and let it down on developers. Despite I'll still try to participate in most circles and helping on guiding stuff forward.

**Comment by damskyftw** [comment]
#### Hey, as usually we've also created space where you can ask questions regarding meta-hours which will be answered asap. Club mate flow.

- #2569

**Comment by vikiival** [comment]
**Viki's Meta Minutes**
-

### DEV in numbers:
`is:pr is:merged created:>=2022-02-16`

Since the last Meta Hours, we merged more than 120 PRs 


and we welcome three new recurrent contributors (@Jarsen136, @preschian, @zhengow)
with actually merged 30 PRs!.



#### NEW

#### wallets
- talisman
- polkadot-js
- subwallet


#### history 

(in collection)

<img width="1404" alt="Screenshot 2022-03-23 at 17 54 17" src="https://user-images.githubusercontent.com/22471030/159753848-6f6a1e90-aa99-41df-90d1-ebb425e62886.png">


#### aero effect


<img width="1400" alt="Screenshot 2022-03-23 at 17 55 47" src="https://user-images.githubusercontent.com/22471030/159754140-3172ad3e-45db-4b2e-a752-91d94847bf04.png">



#### download button

<img width="820" alt="Screenshot 2022-03-23 at 17 56 00" src="https://user-images.githubusercontent.com/22471030/159754201-058a2415-9a77-4e03-adb3-0c11bf322050.png">

#### Writing

SubSquid

**Comment by damskyftw** [comment]
**Damsky pieces** 

Happy to show drafts already submitted to KodaDot publication
- KodaForest article
- Are nfts public good?

How did the KodaForest logo contest go?
- Results
- Summary of this movement

Couple words about community meetups and online workshops 
- Testing out first rounds of this initiative
- Expand

What is Artist workflow initiative?
- Our youtube is growing
- Subwallet tutorial has positive feedback

Get your software ready
- Because animated stickers contest is here

Interesting times ahead - strap in
- Catching up and whats ahead
- Will we meet in Prague in June?

**Comment by petersopko** [comment]
### **Autocode/Cloudflare KV workers as Multipurpose Bot**

Solving two issues:
  - https://github.com/kodadot/nft-gallery/issues/2448
  - https://github.com/kodadot/nft-gallery/issues/1477

Two main functions: 
  
**1. Payouts data collection and Leaderboard**
- Every 10 paid PRs - > open pull request with updated [LEADERBOARD.MD](https://github.com/kodadot/nft-gallery/blob/main/LEADERBOARD.md)
- Payout Object
```
key: prNumber
value: {
  prLeaderboard: boolean, // has been added to LEADERBOARD.MD?
  prAuthor: string,
  prState: string, // merged or closed
  githubLink: string, 
  transactions: [
    {
      transactionResult: string, // Success / Failed
      paidUsd: number,
      paidKsm: number,
      subscanLink: string
      subscanHash: string
     }
   ],
   prCompletionDate:  string,
   commits: number,
   linesAdded: number,
   linesRemoved: number,
   issuesLinked: []
}
```

**2. Assignment and workflow handling**
- Issue Object
```
key: issueNumber,
value: {
  assignee: string, // name of the assignee
  assignmentDeadline: string, // date until assignment is valid
  prOpened: number, // linked PR, set to null in the beginning and changed after PR is open
  queue: []
}
```
- Dev Object
```
key: devGithubLogin,
value: {
  assigned: [], // array of assigned issues, max 5
  finished: [], // 
  unfinished: [],
}
```
- goPhrases: 
```
[
  'I take this',
  'I take this.',
  "I'll take this",
  "I'll take this.",
  'I will take this',
  'I will take this.',
  'i will take this',
  'i take this',
  'i take this.',
  'ill take this',
  'ill take this.',
]
```
![Untitled (2)](https://user-images.githubusercontent.com/73316633/159739818-4768955c-12e0-43f5-9cca-1dba06e8a788.jpg)
